tdesign/src/Layout/BaseLayout/BaseLayout.tsx

```diff
@@ -103,6 +105,7 @@ const BaseLayout = ({
 
   return (
     <Box sx={containerStyle}>
+      {marketingNotice}
```

Comment:
Seeing this component it is starting to look like it can die a death of a thousand new props (the way to modify the layout is to add a new prop, which does not scale well :worried: )

No action required in this PR. I'm just mentioning my concerns for this patterns scalability

The alternative would be to split it into a couple components that can be composed together, and then offer a facade for the most common use-cases. An example separation could be:

* a `HeaderLayout` component (contains the contents of the `Header` JSX from line 110 below)
* a `BaseLayoutContainer` component - `Box sx={containerStyle}`
* `BaseLayout` would use `BaseLayoutContainer` with `HeaderLayout` inside

If you need to render some slightly different layout, you could do:

```tsx
<BaseLayoutContainer>
  <MarketingNotice />
  <HeaderLayout />

  <div>
    The rest of the content here
  </div>
</BaseLayoutContainer>
```

If you wanted to customize the header, you could do:

```tsx
<BaseLayoutContainer>
  <HeaderLayout renderCenter={...} />
  
  <div>Whatever I want</div>
</BaseLayoutContainer>
```

This lets us reduce the number of props (and die due to combinatorial explosion of the number of possibilities) and move the responsibility of determining which layout is wanted to the consumer. Each component would be parametrized instead of having all the props passed to `BaseLayout`.

This is a very general note, again, I'm not expecting any changes here, just wanting to bring up the concern and hear your thoughts on this pattern

Reply:
Not sure I agree – you could just as easily argue that it's premature optimization to add this much composability to every component.

In practice we've hardly edited this layout at all, and the fact that it's separated into `Layout` and `Header` already kind of does what you're suggesting, doesn't it? Not sure I see the difference beyond adding another layer of composability that will likely never be used.

Reply:
> Not sure I agree – you could just as easily argue that it's premature optimization to add this much composability to every component.

Hold on, I didn't suggest every component, I suggested this one because it already receives many props that alter its behavior :smile: 

Maybe it is premature, maybe not :smile: Depending on how many props is too many. If you're saying this rarely gets updated and fits out needs ATM, let's leave it unchanged